src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -410,307 +409,128 @@ public void FormulaCantRedefine()
-        }
-
-        [Fact]
-        public void DefFuncDecimal()
-        {
-            var config = new PowerFxConfig();
-            var recalcEngine = new RecalcEngine(config);
-
-            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("foo(x:Decimal, y:Decimal): Decimal = x * y;").Errors;
-            Assert.False(enumerable.Any());
-            Assert.Equal(17m, recalcEngine.Eval("foo(3,4) + 5").ToObject());
-        }
-
-        [Fact]
-        public void DefFuncWithErrorsAndVerifySpans()
-        {
-            var config = new PowerFxConfig();
-            var recalcEngine = new RecalcEngine(config);
-
-            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("func1(x:Number/*comment*/): Number = x * 10;\nfunc2(x:Number): Number = y1 * 10;").Errors;
-            Assert.True(enumerable.Any());
-            var span = enumerable.First().Span;
-            Assert.Equal(71, span.Min);
-            Assert.Equal(73, span.Lim);
-
-            enumerable = recalcEngine.DefineFunctions("func3():Blank = a + func1(10);").Errors;
-            Assert.True(enumerable.Any());
+        }     
 
-            span = enumerable.First().Span;
-            Assert.Equal(16, span.Min);
-            Assert.Equal(17, span.Lim);
-
-            span = enumerable.ElementAt(1).Span;
-            Assert.Equal(20, span.Min);
-            Assert.Equal(29, span.Lim);
-        }
-
-        [Fact]
-        public void DefRecursiveFunc()
-        {
-            var config = new PowerFxConfig();
-            var recalcEngine = new RecalcEngine(config);
-            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("foo(x:Number):Number = If(x=0,foo(1),If(x=1,foo(2),If(x=2,Float(2)));", numberIsFloat: true).Errors;
-            var result = recalcEngine.Eval("foo(Float(0))");
-            Assert.Equal(2.0, result.ToObject());
-            Assert.False(enumerable.Any());
-        }
-
-        [Fact]
-        public void DefRecursiveFuncDecimal()
-        {
-            var config = new PowerFxConfig();
-            var recalcEngine = new RecalcEngine(config);
-            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("foo(x:Decimal):Decimal = If(x=0,foo(1),If(x=1,foo(2),If(x=2,2));").Errors;
-            var result = recalcEngine.Eval("foo(0)");
-            Assert.Equal(2m, result.ToObject());
-            Assert.False(enumerable.Any());
-        }
-
-        [Fact]
-        public void DefSimpleRecursiveFunc()
-        {
-            var config = new PowerFxConfig();
-            var recalcEngine = new RecalcEngine(config);
-            Assert.False(recalcEngine.DefineFunctions("foo():Blank = foo();").Errors.Any());
-            var result = recalcEngine.Eval("foo()");
-            Assert.IsType<ErrorValue>(result);
-        }
-
-        [Fact]
-        public void DefHailstoneSequence()
+        [Theory]

```

Comment:
>Theory [](http://example.com/codeflow?start=9&length=6)

Can we add test, demoing chaining of different udfs?
e.g. Odd(foo())

Reply:
The test case `DateT3333333333333imeSerializationTests` (at SerializationTests) had a similar test case. This test was left by mistaken tough but I will include another on to the new txt files.